operations/helm/charts/grafana-agent/templates/hpa.yaml

```diff
@@ -1,4 +1,16 @@
 {{- if and (or (eq .Values.controller.type "deployment") (eq .Values.controller.type "statefulset" ))  .Values.controller.autoscaling.enabled }}
+{{- if not (empty .targetMemoryUtilizationPercentage) }} 
```

Comment:
Initially, I just checked for the presence of the resources.requests field, but then added the more fine-grained checks.

If you think this is too much, I can revert back to the simpler check with a better error message.